### PR TITLE
Feature/identity

### PR DESCRIPTION
**Pull request by Sokel**

### CODE REVIEW

File: accounts/identity.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+// Load keystore
+// this implementation
+func (idt *identityPassphrase) Load(keystoreDir *string) (err error) {
+	if keystoreDir == nil {
+		homeDir, err := util.GetUserHomeDir()
+		if err != nil {
+			return err
+		}
+		*keystoreDir = filepath.Join(homeDir, "sonm", "keystore")
```
Comment (noxiouz) [inline accounts/identity.go]: https://play.golang.org/p/j3c4Kvr2_S

File: accounts/identity.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+	if err != nil {
+		return err
+	}
+	idt.key = key
+	idt.privateKey = key.PrivateKey
+	return nil
+}
+
+func parseKeystoreUrl(url string) ([]string, error) {
+	parts := strings.Split(url, "://")
```
Comment (noxiouz) [inline accounts/identity.go]: url.Parse?

File: util/util.go [hunk 3]
```diff
@@ -57,3 +58,11 @@ func GetPublicIP() (net.IP, error) {
 	}
 	return pubipadr, nil
 }
+
+func GetUserHomeDir() (homeDir string, err error) {
```
Comment (noxiouz) [inline util/util.go]: no point to have named return vars

File: accounts/identity.go [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+		*keystoreDir = filepath.Join(homeDir, "sonm", "keystore")
+	}
+	err = idt.initKeystore(keystoreDir)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (idt *identityPassphrase) Open(pass string) (err error) {
```
Comment (noxiouz) [inline accounts/identity.go]: Identity interface requires Open(*string)

File: accounts/identity.go [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+	return idt.readPrivateKey(pass)
+}
+
+func (idt *identityPassphrase) GetPrivateKey() (*ecdsa.PrivateKey, error) {
+	if idt.privateKey == nil {
+		return nil, errors.New("Wallet doesn't opened now")
```
Comment (noxiouz) [inline accounts/identity.go]: is not open

File: accounts/identity.go [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+	key, err := keystore.DecryptKey(file, pass)
+	if err != nil {
+		return err
+	}
+	idt.key = key
+	idt.privateKey = key.PrivateKey
+	return nil
+}
+
+func parseKeystoreUrl(url string) ([]string, error) {
```
Comment (jorpic) [inline accounts/identity.go]: Returning array of 2 elements is not quite declarative.
I think `parseKeystoreUrl(url string) (<descriptive_name_of_first_part> string, <descriptive_name_of_second_part> string, error)` will be less error prone and much easier to understand.

File: accounts/identity.go [hunk 7]
```diff
@@ -99,7 +75,30 @@ func (idt *identityPassphrase) initKeystore(keydir string) (err error) {
...
 	}
 	idt.defaultAccount = accs[0]
-	return nil
+
+	return idt.readPrivateKey(pass)
+}
+
+func (idt *identityPassphrase) GetPrivateKey() (*ecdsa.PrivateKey, error) {
+	if idt.privateKey == nil {
+		return nil, errors.New("Wallet is not open now")
```
Comment (sshaman1101) [inline accounts/identity.go]: typed errors?

File: accounts/identity.go [hunk 8]
```diff
@@ -99,7 +75,30 @@ func (idt *identityPassphrase) initKeystore(keydir string) (err error) {
...
+	}
+	return idt.privateKey, nil
+}
+
+func (idt *identityPassphrase) New(passphrase string) (err error) {
```
Comment (sshaman1101) [inline accounts/identity.go]: I think there is no reason to use named return values

**Pull request closed without merging**